### PR TITLE
fix!: `#app` module augmentation and types

### LINKED ISSUE #150: Weird vue router warnings on cold start
Thanks for trying to tackle all these Nuxt<>Vuetify issues with this library.

I pretty much copied the playground setup with the configFile variant, because I want to customize the components.
It works fine, the hydration time and cold start time is a bit longer, but that's normal, as Vuetify docs said.

However, the server and client logs print this:
```log
ℹ Vite server warmed up in 1447ms                                             4:21:32 PM
ℹ Vite client warmed up in 1668ms                                             4:21:32 PM
✔ Nitro built in 352 ms                                                 nitro 4:21:32 PM
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VAppBar/VAppBar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VAppBar/VAppBar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VAppBar/VAppBar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VAppBar/VAppBar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VImg/VImg.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VImg/VImg.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VImg/VImg.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VImg/VImg.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VApp/VApp.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VApp/VApp.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VApp/VApp.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VApp/VApp.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VToolbar/VToolbar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VToolbar/VToolbar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VToolbar/VToolbar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VToolbar/VToolbar.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtn/VBtn.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtn/VBtn.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtn/VBtn.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtn/VBtn.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtnGroup/VBtnGroup.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtnGroup/VBtnGroup.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtnGroup/VBtnGroup.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VBtnGroup/VBtnGroup.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VIcon/VIcon.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VIcon/VIcon.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VIcon/VIcon.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VIcon/VIcon.sass"
[Vue Router warn]: No match found for location with path "/_nuxt/@plugin-vuetify/lib/components/VResponsive/VResponsive.sass"
... many more components :)
```

Just thought I'd document it, it's not really causing any issues (as far as I can tell).
Almost seems its the same issue we had to patch in the nitro response, except now its in vue router...

### PR DESCRIPTION
**Pull request by userquin**
This PR includes:
- update module meta for Nuxt 3.6.5+ and no bridge
- revert `#app/nuxt` module augmentation to `#app`
- reverted package exports: node16-esm and node16-cjs cannot resolve module imports without extension in d.ts and d.mts files
- docs: update some Nuxt texts, some imports in code snippets and added i18n hint to use latest edge  release
- renamed all mts files to ts
- renaming node configuration types
- bump to Vuetify ~~^3.4.0~~^3.4.3: logic updated for labs components and `useDate` composable and fixing 3.4.0 `VDatePickerYears.tsx:77 Uncaught (in promise) TypeError: adapter.getNextYear is not a function`
- bump to VP rc.29 and VP PWA integration ^0.17.0 (Vite 5)
- bump to TypeScript ^5.3.2
- **NOTE**: `useNuxtApp().$i18n` will resolve the type properly only when using `Node` module resolution, using `Bundler` the type is `any`

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -72,3 +79,38 @@ export default defineNuxtModule<ModuleOptions>({
...
+  }) => HookResult
+  'vuetify:ready': (vuetify: ReturnType<typeof createVuetify>) => HookResult
+  'vuetify:ssr-client-hints': (options: {
+    vuetifyOptions: VuetifyOptions
+    ssrClientHints: SSRClientHints
+    ssrClientHintsConfiguration: SSRClientHintsConfiguration
+  }) => HookResult
+}
+
+declare module '#app/nuxt' {
```
Comment (userquin) [inline src/module.ts:104]: should be `#app`: https://github.com/nuxt/nuxt/issues/24193#issuecomment-1809959962

File: package.json [hunk 2]
```diff
@@ -63,43 +64,43 @@
     "release": "bumpp && npm publish"
   },
   "dependencies": {
-    "@nuxt/kit": "^3.6.2",
-    "defu": "^6.1.2",
+    "@nuxt/kit": "^3.8.1",
```
Comment (userquin) [inline package.json:67]: check also to revert this and check it with i18n edge version (should be using `#app`)

File: src/module.ts [hunk 3]
```diff
@@ -23,7 +30,7 @@ export default defineNuxtModule<ModuleOptions>({
   meta: {
     name: 'vuetify-nuxt-module',
     configKey: 'vuetify',
-    compatibility: { nuxt: '^3.0.0' },
+    compatibility: { nuxt: '^3.8.1' },
```
Comment (userquin) [inline src/module.ts:33]: revert also this

Comment (userquin) [reply src/module.ts]: changed to ^3.6.5

File: package.json [hunk 4]
```diff
@@ -21,29 +21,29 @@
   ],
   "exports": {
     ".": {
-      "types": "./dist/types.d.ts",
-      "require": "./dist/module.cjs",
-      "import": "./dist/module.mjs"
+      "types": "./dist/module.d.mts",
```
Comment (danielroe) [inline package.json:24]:
```suggestion
      "types": "./dist/types.d.mts",
```

File: package.json [hunk 5]
```diff
@@ -21,29 +21,29 @@
...
       "types": "./configuration.d.ts"
-    }
+    },
+    "./*": "./*"
   },
-  "main": "./dist/module.cjs",
-  "types": "./dist/types.d.ts",
+  "main": "./dist/module.mjs",
+  "module": "./dist/module.mjs",
+  "types": "./dist/module.d.mts",
```
Comment (danielroe) [inline package.json:38]:
```suggestion
  "types": "./dist/types.d.mts",
```

**Review by sctnightcore — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -21,20 +21,20 @@
...
     "./configuration": {
       "types": "./configuration.d.ts"
-    }
+    },
+    "./*": "./*"
   },
-  "main": "./dist/module.cjs",
+  "main": "./dist/module.mjs",
+  "module": "./dist/module.mjs",
   "types": "./dist/types.d.ts",
```
Comment (userquin) [inline package.json:38]: uhmm

Comment (userquin) [reply package.json:38]: should be ESM only, using module.cjs will break Node10 module resolution

Comment (userquin) [reply package.json:38]: this is fine

**Pull request merged**